include/swift/AST/DiagnosticsSema.def

```diff
@@ -2435,6 +2435,50 @@ NOTE(fix_unqualified_access_top_level_multi,none,
 ERROR(unsupported_special_decl_ref, none,
       "referencing %0 as a function value is not implemented", (Identifier))
 
+WARNING(bitcasting_away_noescape, none,
+        "'unsafeBitCast' from nonescaping function type %0 to escaping "
+        "function type %1 is undefined; use "
+        "'withoutActuallyEscaping' to temporarily escape a function",
+        (Type, Type))
+WARNING(bitcasting_to_change_function_rep, none,
+        "'unsafeBitCast' from function type %0 to %1 changes @convention and "
+        "is undefined; use an implicit conversion to change conventions",
+        (Type, Type))
+WARNING(bitcasting_to_downcast, none,
+        "'unsafeBitCast' from %0 to %1 can be replaced with 'unsafeDowncast'",
+        (Type, Type))
+WARNING(bitcasting_is_no_op, none,
+        "'unsafeBitCast' from %0 to %1 is a no-op",
```

Comment:
Maybe, but I couldn't think of a well-worded alternative. What would you suggest?